discovery/gossiper.go

```diff
@@ -2291,42 +2291,15 @@ func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
 	ann *lnwire.ChannelAnnouncement,
 	ops []batch.SchedulerOption) ([]networkMsg, bool) {
 
-	log.Debugf("Processing ChannelAnnouncement: peer=%v, short_chan_id=%v",
-		nMsg.peer, ann.ShortChannelID.ToUint64())
-
-	// We'll ignore any channel announcements that target any chain other
-	// than the set of chains we know of.
-	if !bytes.Equal(ann.ChainHash[:], d.cfg.ChainHash[:]) {
-		err := fmt.Errorf("ignoring ChannelAnnouncement from chain=%v"+
-			", gossiper on chain=%v", ann.ChainHash,
-			d.cfg.ChainHash)
-		log.Errorf(err.Error())
+	// jobDesc is used to verbosely log the ChannelAnnouncement.
+	jobDesc := fmt.Sprintf("ChannelAnnouncement: peer=%v, "+
+		"short_chan_id=%v", nMsg.peer, ann.ShortChannelID.ToUint64())
 
-		key := newRejectCacheKey(
-			ann.ShortChannelID.ToUint64(),
-			sourceToPub(nMsg.source),
-		)
-		_, _ = d.recentRejects.Put(key, &cachedReject{})
-
-		nMsg.err <- err
-		return nil, false
-	}
-
-	// If this is a remote ChannelAnnouncement with an alias SCID, we'll
-	// reject the announcement. Since the router accepts alias SCIDs,
-	// not erroring out would be a DoS vector.
-	if nMsg.isRemote && d.cfg.IsAlias(ann.ShortChannelID) {
-		err := fmt.Errorf("ignoring remote alias channel=%v",
-			ann.ShortChannelID)
-		log.Errorf(err.Error())
-
-		key := newRejectCacheKey(
-			ann.ShortChannelID.ToUint64(),
-			sourceToPub(nMsg.source),
-		)
-		_, _ = d.recentRejects.Put(key, &cachedReject{})
+	log.Debugf("Processing %s", jobDesc)
 
-		nMsg.err <- err
+	// Validate the channel announcement.
+	if err := d.validateChannelAnnouncement(nMsg, ann); err != nil {
+		log.Errorf("Validating %s failed %v", jobDesc, err)
```

Comment:
Looks to no longer be sending the error back over `nMsg.err` ?

Reply:
Changed it to send the error to make the flow more clear.